orderbook/src/orderbook.rs

```diff
@@ -126,10 +127,13 @@ impl Orderbook {
         ) {
             return Ok(AddOrderResult::UnsupportedSignature);
         }
-
         if has_same_buy_and_sell_token(&order, &self.native_token) {
             return Ok(AddOrderResult::SameBuyAndSellToken);
         }
+        if order.buy_amount.is_zero() || order.sell_amount.is_zero() {
```

Comment:
In theory, `buy_amount` can be 0 for a "true" market order. Although, for all practical purposes `1` is good enough and avoids sneaky divide by 0 issues that may come up.